### PR TITLE
add plivo webhook parser

### LINKED ISSUE #1464: [NV-1016] - Create webhook parser for Plivo (sms)
**Why**

To be able to give better answers why a message was not sent to a subscriber we need to get some info from the provider with a webhook. (deliverability status, rejection status etc).

**How**

To do this we need to implement these functions inside of the provider:

```
  getMessageId?: (body: any | any[]) => string[];

  parseEventBody?: (
    body: any | any[],
    identifier: string
  ) => IEmailEventBody | undefined;
```

The `getMessageId` needs to return all identifiers that should be the same value as id that is returned from `sendMessage` with help of this we can connect a webhook to the specific message.

`parseEventBody` will get the full webhook body and the current identifier that the body should be parsed for. If there is nothing to parse the method should return `undefined`. 

All returns that is not `undefined` should be an object that is in this format:

```
status: string;

 date: string;

externalId?: string;

attempts?: number;

response?: string;

// Contains the raw content for message from the provider webhook

row?: string;
```

Note that statuses for email needs to be value from `EmailEventStatusEnum` and sms should have a status from `SmsEventStatusEnum.` 

 Example of how to do it can be found in `SendgridEmailProvider`.

### PR DESCRIPTION
**Pull request by mocherfaoui**
### What change does this PR introduce?

added Plivo webhook parser
### Why was this change needed?

closes #1464 

----------------
I have a question though, the Novu webhook URL is not specified when [sending a message](https://www.plivo.com/docs/sms/api/message#send-a-message) with Plivo?
the same case for Twilio, there is no `statusCallback` ?

### CODE REVIEW

File: providers/plivo/src/lib/plivo.provider.ts [hunk 1]
```diff
@@ -32,8 +34,61 @@ export class PlivoSmsProvider implements ISmsProvider {
...
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body.messageUuid,
+      attempts: body.attempt ? parseInt(body.attempt, 10) : 1,
+      response: body.response ? body.response : '',
```
Comment (p-fernandez) [inline providers/plivo/src/lib/plivo.provider.ts:73]:
```suggestion
      response: body.response ?? '',
```

**Review by p-fernandez — APPROVED** [review]
🌟

**Comment by mocherfaoui** [comment]
@p-fernandez can you check if my logic is correct? 
plivo returns `messageUuid` as an array and it's better than using `apiId` as an id since the webhook callback return `messageUuid` only?

File: providers/plivo/src/lib/plivo.provider.ts [hunk 2]
```diff
@@ -32,8 +34,61 @@ export class PlivoSmsProvider implements ISmsProvider {
...
+    const status = this.getStatus(body.status);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body.messageUuid,
```
Comment (p-fernandez) [inline providers/plivo/src/lib/plivo.provider.ts:71]: This looks ok. 👍🏻

**Review by p-fernandez — COMMENTED** [review]
I'd encourage you to add tests. 🙏🏻

**Comment by mocherfaoui** [comment]
@p-fernandez there is a plivo test in place, if you can approve providers' tests the plivo test will pass this time.

**Review by scopsy — APPROVED** [review]

**Pull request merged**